### PR TITLE
chore: extend enum endpoint

### PR DESCRIPTION
**Pull request by juliapampus**

### CODE REVIEW

File: src/main/java/io/dataspaceconnector/controller/gui/util/GuiUtils.java [hunk 1]
```diff
@@ -43,61 +51,38 @@ private GuiUtils() {
...
+        }};
     }
 
     private static JSONArray getPaymentMethod() {
         final var jsonArray = new JSONArray();
         for (final var paymentMethod : PaymentMethod.values()) {
-            try {
-                jsonArray.add(new JSONObject() {{
-                    put("originalName", paymentMethod.name());
-                    put("displayName", PaymentMethod.class.getField(paymentMethod.name())
```
Comment (ghost) [inline src/main/java/io/dataspaceconnector/controller/gui/util/GuiUtils.java:92]: I am not sure if removing the JsonProperty-content might not cause problems. I think that the PUT and POST APIs from GUI to DSC need it in the JSON.

e.g. 
@JsonProperty("**fixedPrice**")
FIXED_PRICE("Fixed price")

With the adjustment only the two lower enum-name/value would be output `FIXED_PRICE` (by paymentMethod.name() as originalName) and `Fixed price` (by paymentMethod.toString() as displayName) but not `fixedPrice`. Same for negotiationBasis. 

I could be wrong, didn't runtime test it.

**Review by brianjahnke — CHANGES_REQUESTED** [review]
Hi,
- The test coverage does not seem high enough
- The openapi yaml needs an update
- Not all changes are documented in the changelog?

**Comment by juliapampus** [comment]
> Hi,
> 
> * The test coverage does not seem high enough
> * The openapi yaml needs an update
> * Not all changes are documented in the changelog?

OpenApi will be changed anyway in PR #838. What changes are not in the `CHANGELOG.md`? 

Edit: openapi was updated

**Comment by brianjahnke** [comment]
Hi,
`Not all changes are documented in the changelog?`

I think the enums changed to some string. For example MY_DATA to MyData.
I was not sure if this needs to be documented therefore the `?`.

**Review by brianjahnke — DISMISSED** [review]

**Comment by ghost** [comment]
wait for me, still checking something

**Comment by brianjahnke** [comment]
ok :suspect:

**Review by brianjahnke — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**